pkg/index/index.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 The Falco Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Entry describes an entry of the index stored remotely and cached locally.
+type Entry struct {
+	Description string   `yaml:"description"`
+	Home        string   `yaml:"home"`
+	Keywords    []string `yaml:"keywords"`
+	License     string   `yaml:"license"`
+	Maintainers []struct {
+		Email string `yaml:"email"`
+		Name  string `yaml:"name"`
+	} `yaml:"maintainers"`
+	Name       string   `yaml:"name"`
+	Registry   string   `yaml:"registry"`
+	Repository string   `yaml:"repository"`
+	Sources    []string `yaml:"sources"`
+	Type       string   `yaml:"type"`
+}
+
+// Index represents an index.
+type Index struct {
+	Name        string
+	Filename    string
+	Entries     []*Entry
+	entryByName map[string]*Entry
+}
+
+// MergedIndexes is used to aggregate all indexes and perform search operations.
+type MergedIndexes struct {
+	Index
+	indexByEntry map[*Entry]*Index
+}
+
+// NewIndex loads an Index from a file.
+func NewIndex(path, name string) (*Index, error) {
```

Comment:
```suggestion
func New(path, name string) (*Index, error) {
```

We are in the `index` package, so `index.New()` is preferable.